projectile.el

```diff
@@ -6155,8 +6155,11 @@ when opening new files."
 ;; it's safe to require this directly, as it was added in Emacs 25.1
 (require 'project)
 
-(cl-defmethod project-root ((project (head projectile)))
-  (cdr project))
+;; Only define an override for project-root if the method exists.  For versions
+;; before emacs 28, project.el provided project-roots instead of project.root.
+(if (fboundp 'project-root)
```

Comment:
Shouldn't we override the old method in the else branch for completeness?

Reply:
Probably, though I haven't directly used the integration, and so I wouldn't have been able to verify that the changes work as intended.  In principle, the overload would just be the overload for `project-root`, but returning a list of length 1 instead of a single element.

```lisp
(cl-defmethod project-roots ((project (head projectile)))
      (list (cdr project))))
```